arrow-libs/core/arrow-core/src/main/kotlin/arrow/core/map.kt

```diff
@@ -224,6 +224,12 @@ inline fun <K, E, A, B> Map<K, A>.traverseValidated(
 fun <K, E, A> Map<K, Validated<E, A>>.sequenceValidated(semigroup: Semigroup<E>): Validated<E, Map<K, A>> =
   traverseValidated(semigroup, ::identity)
 
+inline fun <K, A, B> Map<K, A>.traverseOption(f: (A) -> Option<B>): Option<Map<K, B>> =
+  traverseEither { f(it).toEither { Unit } }.orNone()
```

Comment:
```suggestion
inline fun <K, A, B> Map<K, A>.traverseOption(f: (A) -> Option<B>): Option<Map<K, B>> {
  val acc = mutableMapOf<K, B>()
  forEach { (k, v) ->
    when (val res = f(v)) {
      is Some -> acc[k] = res.value
      None -> return@traverseOption res
    }
  }
  return acc.some()
}
```


Nit: a direct implementation reads better IMO, and is (slightly) more performant.
That'll save 1 allocation per value pair in the map.